### PR TITLE
Moved Navigated before OnNavigatedTo and OnNavigatedFrom

### PR DESCRIPTION
**Pull request by DZetko**
GitHub Issue (If applicable): #1863 

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Wrong order of navigation events.

## What is the new behavior?

Correct order of navigation events.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by jeromelaban** [comment]
Thanks for your contribution!

We do have unit tests for `Frame` here:

https://github.com/unoplatform/uno/blob/47a749468863e4e9a65dae6d3a56714444491d35/src/Uno.UI.Tests/Windows_UI_XAML_Controls/Frame/Given_Frame.cs#L19

Could you add one to validate the behavior and sequence?

File: src/Uno.UI/UI/Xaml/Controls/Frame/Frame.cs [hunk 1]
```diff
@@ -326,9 +326,9 @@ private bool InnerNavigate(PageStackEntry entry, NavigationMode mode)
 					null
 				);
 
+				Navigated?.Invoke(this, navigationEvent);
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Frame/Frame.cs]: Events most likely have to be moved before `Content = CurrentEntry.Instance;` above for `x:Bind` to done after  navigatedto properly.

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Frame/Frame.cs]: Unfortunately, it must stay after setting the `Content` - in UWP `Content` is already set when `Navigated` is raised. It seems the `Loading` and `Loaded` events need to be initiated by the frame when the page is inside of it.

```
Frame navigating
Page constructor
Frame.Content == EventOrdering.MainPage
Frame navigated
Page navigated to
Page loading
Page loaded
```

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Frame/Frame.cs]: @jeromelaban I have created #2895 for it. It seems it will be a very complex issue.

**Comment by jeromelaban** [comment]
Let's close this PR for now as it will be reworked by the Loaded/Unloaded work.

**Pull request closed without merging**